src/framework/ui/qml/MuseScore/Ui/internal/StandardDialogPanel.qml

```diff
@@ -112,126 +107,110 @@ Rectangle {
         name: root.title + " " + root.text + " " + root.firstFocusBtn().text + " " + qsTrc("global", "Button")
     }
 
-    ColumnLayout {
-        id: content
-
-        anchors.fill: parent
-
-        property int contentHeight: childrenRect.height
-
-        spacing: 16
-
-        RowLayout {
-            id: textContent
-
-            property int contentWidth: (icon.visible ? icon.width + textContent.spacing : 0) + textColumn._preferredWidth
-
-            Layout.preferredWidth: content.width
-            Layout.preferredHeight: childrenRect.height
-
-            spacing: 28
-
-            StyledIconLabel {
-                id: icon
-
-                Layout.alignment: Qt.AlignTop
-                Layout.preferredWidth: 48
-                Layout.preferredHeight: 48
+    spacing: 27
 
-                font.pixelSize: 48
-                iconCode: root.standardIcon(root.type)
+    StyledIconLabel {
+        id: icon
 
-                visible: root.withIcon && !isEmpty
-            }
-
-            Column {
-                id: textColumn
+        Layout.alignment: Qt.AlignTop
+        Layout.preferredWidth: 48
+        Layout.preferredHeight: 48
 
-                property int _preferredWidth: Math.max(Math.min(titleLabel.implicitWidth, 420),
-                                                       Math.min(textLabel.implicitWidth, 420))
+        font.pixelSize: 48
+        iconCode: root.standardIcon(root.type)
 
-                Layout.preferredWidth: _preferredWidth + /*todo*/ 76
-                Layout.preferredHeight: childrenRect.height
+        visible: root.withIcon && !isEmpty
+    }
 
-                spacing: 16
+    Column {
+        Layout.fillWidth: true
 
-                StyledTextLabel {
-                    id: titleLabel
+        // Unclear why "+ 1" is needed; apparently implicitWidth is incorrect
+        readonly property real textsImplicitWidth: Math.max(titleLabel.implicitWidth,
+                                                            textLabel.implicitWidth,
+                                                            withShowAgainCheckBox.implicitWidth) + 1
 
-                    width: parent._preferredWidth + /*todo*/ 4
+        // Allow the text to make the dialog wider, but not too much wider
+        readonly property real textsImplicitWidthBounded: Math.min(420, textsImplicitWidth)
 
-                    font: ui.theme.largeBodyBoldFont
-                    horizontalAlignment: Text.AlignLeft
-                    maximumLineCount: 2
-                    wrapMode: Text.Wrap
+        // But if the buttons need more space, then the dialog becomes as wide as necessary
+        Layout.preferredWidth: Math.max(buttons.width, textsImplicitWidthBounded)
```

Comment:
Note: I came up with this "algorithm" myself. I did not manage to find a pattern that is followed by all examples on Figma. In practice, the results of this computation look good to me.